### PR TITLE
[Method References 2] LPS-165626

### PR DESCRIPTION
**Pull request by JanisZhang**
Resend.
Previous pr:https://github.com/liferay-core-infra/liferay-portal/pull/2239

### CODE REVIEW

**Comment by Kvale1733** [comment]
ci:test:echo

**Comment by Kvale1733** [comment]
ci:test:echo

**Comment by zhangyan0701** [comment]
No regressions found.

File: modules/apps/journal/journal-service/src/main/java/com/liferay/journal/internal/transformer/DefaultJournalTransformerListenerRegistryImpl.java [hunk 1]
```diff
@@ -54,28 +54,26 @@ public List<TransformerListener> getTransformerListeners() {
...
-	public void registerTransformerListener(
-		TransformerListener transformerListener) {
+	@Activate
+	protected void activate(BundleContext bundleContext) {
+		_serviceTrackerMap = ServiceTrackerMapFactory.openSingleValueMap(
+			bundleContext, TransformerListener.class,
+			"(javax.portlet.name=" + JournalPortletKeys.JOURNAL + ")",
+			(serviceReference, emitter) -> {
+				TransformerListener transformerListener =
+					bundleContext.getService(serviceReference);
```
[the remark this answers is not in the document]
Comment (dantewang) [reply modules/apps/journal/journal-service/src/main/java/com/liferay/journal/internal/transformer/DefaultJournalTransformerListenerRegistryImpl.java]: @JanisZhang @W2jx Sorry mentioned the wrong person😂

[the remark this answers is not in the document]
Comment (JanisZhang) [reply modules/apps/journal/journal-service/src/main/java/com/liferay/journal/internal/transformer/DefaultJournalTransformerListenerRegistryImpl.java]: @dantewang Got it, thank you!

**Comment by JanisZhang** [comment]
Hi @dantewang, Force-pushed, and changed to use ServiceReferenceMapperFactory. Thanks.

**Comment by JanisZhang** [comment]
ci:test:sf

**Comment by JanisZhang** [comment]
ci:test:relevant

**Comment by catarineaguiar** [comment]
@JanisZhang no regressions found.

**Comment by JanisZhang** [comment]
@catarineaguiar thx

**Comment by tinatian** [comment]
@JanisZhang Why you need to do "Move JournalTransformerListenerRegistryUtil from module journal-service to journal-api"? All the usages are in the same module now, there is no need to expose it other modules.

**Comment by tinatian** [comment]
@JanisZhang @dantewang There is only one usage of this method "public TransformerListener getTransformerListener(String className);", we should be able to remove it, then you won't need a ServiceTrackerMap, a ServiceTrackerList is enough.

Tina

**Comment by JanisZhang** [comment]
> 

Hi @tinatian , The test need to use `JournalTransformerListenerRegistryUtil`

**Comment by dantewang** [comment]
@JanisZhang Hi Janis, please don't change any portal runtime code for test usages. We should always try best to avoid them.

On the other hand, which test/commit are you mentioning? Slack me and let's see if we can find another way to deal with it.

**Pull request closed without merging**